subsys/net/lib/http/Kconfig

```diff
@@ -106,18 +106,26 @@ config HTTP_SERVER_MAX_HEADER_LEN
 	  internal header processing, and only needs to be increased if the
 	  application wishes to access headers of a greater length.
 
+config HTTP_SERVER_CAPTURE_HEADERS
+	bool "Allow capturing HTTP headers for application use"
+	help
+	  This setting enables the HTTP server to capture selected headers that have
+	  been registered by the application.
+
 config HTTP_SERVER_CAPTURE_HEADER_BUFFER_SIZE
 	int "Size of buffer for capturing HTTP headers for application use"
-	default 0
-	range 0 2048
+	default 128
+	range 32 2048
+	depends on HTTP_SERVER_CAPTURE_HEADERS
 	help
 	  This setting determines the size of the (per-client) buffer used to store
 	  HTTP headers for later use by the application.
```

Comment:
We could leave allocation of this buffer to the application which could pass information to the server somehow, then we could just store the pointer to the user supplied buffer and its size. This way we do not need to manage and possibly over allocate the buffer if the server connection does not need so much of it.

Reply:
I thought it was simpler to allocate in the server - the buffer only exists if the application has enabled `CONFIG_HTTP_SERVER_CAPTURE_HEADERS`, and if the application enables this the resource usage will be roughly the same whether the buffers are allocated on the server side (with a size configured by application via Kconfig) or if the application allocates the same size buffers and passes them in.